lib/icinga/macroprocessor.cpp

```diff
@@ -376,13 +376,17 @@ void MacroProcessor::ValidateCustomVars(const ConfigObject::Ptr& object, const D
 }
 
 void MacroProcessor::AddArgumentHelper(const Array::Ptr& args, const String& key, const String& value,
-	bool add_key, bool add_value)
+	bool add_key, bool add_value, const Value& sep)
 {
-	if (add_key)
-		args->Add(key);
+	if (sep.GetType() == ValueEmpty || !(add_key && add_value)) {
```

Comment:
Other places in the code simply compare with `x == Empty`, shouldn't this be just fine here as well?

I think swapping the cases and using `if (add_key && add_value && sep != Empty)` would be easier to understand, i.e. "we're adding both and there's an explicit separator, so do that special case, otherwise add them separately."

Reply:
* `git grep '[!=]=\s*Empty'`: 74 matches, mostly in one test though (2 matches in `lib/`)
* `git grep 'GetType()\s*[!=]=\s*ValueEmpty'`: 3 matches in total
* `git grep 'IsEmpty()'`: definitely the most common check for emptiness though.

But in summary, that check is done so that `sep = ""` concatenates key and value without anything in between?